👨‍🔬 Beta Workflow to Modify Ploidy

migbro opened the pull request:
<!--Pull Request Template-->

## Description

This started out as a research question for trisomy 21 in which we ran a version of this workflow for a collaborator to review. Since then, another collaborator has brought up a more general case of treating chr X as haploid when the sample is from a male. We achieve this by:
 - Using exiting cram nd GVCF as inputs
 - Subset cram on desired chomosome
 - Use a desired re-calling interval list to limit region calls to...
 - Added ploidy input

Fixes # https://d3b.atlassian.net/browse/BIXU-3785

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] https://cavatica.sbgenomics.com/u/d3b-bixu/kf-germline-harmonization-dev/tasks/cb72626c-50dd-4a50-9c35-9938320fdfe6/
- [ ] Test B

**Test Configuration**:
* Environment:
* Test files:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [x] I have committed any related changes to the PR

migbro commented on tools/samtools_cram_to_bam.cwl:
```diff
@@ -8,19 +8,24 @@ requirements:
+  input_cram: { type: File, secondaryFiles: ['.crai?'], doc: "cram file to convert",
+    inputBinding: { position: 2 } }
+  region: { type: 'string?', doc: "Specific region to pull, in format 'chr21' or 'chr3:1-1000'",
+    inputBinding: { position: 2 } }
```
technically it does, because the tie breaker of them both being position 2 is that I declared input cram first. However, that is a dangerous technicality, so I'll make it 1

sickler-alex reviewed (commented):
Looks good. Holding off on approval until all of Dan's comments are addressed.

migbro commented:
@dmiller15 , ok, made a tool to address the header issue. Runs pretty quick: https://cavatica.sbgenomics.com/u/d3b-bixu/kf-germline-harmonization-dev/tasks/be613e5e-34e4-4590-816e-375afc43b266/

migbro commented:
Implemented and modified @dmiller15 suggestion. Test here: https://cavatica.sbgenomics.com/u/d3b-bixu/kf-germline-harmonization-dev/tasks/45766bae-9c4f-462f-8c3a-13fb0644ae46/

dmiller15 reviewed (approved):
Nice work
